LookupTable.lua

```diff
@@ -95,6 +108,25 @@ function LookupTable:accGradParameters(input, gradOutput, scale)
    )
 end
 
+function LookupTable:renorm(input)
+   if not self.maxNorm then
+      return
+   end
+   local row_idx = input:clone()
```

Comment:
`:clone()` allocates memory every forward pass. This can cause performance problems on the GPU because the corresponding GC/cuda_free's will cause synchronization points.

Instead, re-use a temporary buffer (like in `makeInputContiguous`)